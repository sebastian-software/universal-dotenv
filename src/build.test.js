/* eslint-disable @typescript-eslint/no-var-requires */
process.env.ENV_CONTEXT = "client"

const snapshotOpts = {
  APP_ROOT: expect.any(String),
  APP_SOURCE: expect.any(String)
}

const savedProcessEnv = { ...process.env }
afterAll(() => {
  process.env = savedProcessEnv
})

/* eslint-disable import/no-commonjs */
// We can't use ESM when relying on the fact the the env from the top is correctly respected.
const api = require("..")

api.init()

test("Serializes ENV_CONTEXT", () => {
  const { raw, stringified, webpack } = api.getEnvironment()
  expect(raw).toMatchSnapshot(snapshotOpts)
  expect(stringified).toMatchSnapshot(snapshotOpts)
  expect(webpack).toBeDefined()
})

test("Exports ENV_CONTEXT", () => {
  const { raw } = api.getEnvironment()
  expect(raw.ENV_CONTEXT).toBe("client")
})
