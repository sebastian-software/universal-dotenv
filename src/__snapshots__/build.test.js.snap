// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Serializes ENV_CONTEXT 1`] = `
Object {
  "APP_ROOT": Any<String>,
  "APP_SOURCE": Any<String>,
  "APP_TEST_IS_CLIENT": true,
  "APP_TEST_OVERRIDE": "overridden",
  "APP_TEST_PATH": "one/two/three.js",
  "ENV_CONTEXT": "client",
  "NODE_ENV": "test",
}
`;

exports[`Serializes ENV_CONTEXT 2`] = `
Object {
  "APP_ROOT": Any<String>,
  "APP_SOURCE": Any<String>,
  "APP_TEST_IS_CLIENT": "true",
  "APP_TEST_OVERRIDE": "\\"overridden\\"",
  "APP_TEST_PATH": "\\"one/two/three.js\\"",
  "ENV_CONTEXT": "\\"client\\"",
  "NODE_ENV": "\\"test\\"",
}
`;
